src/libs/decoders/SDL_sound.c

```diff
@@ -145,7 +145,10 @@ int Sound_Quit(void)
     BAIL_IF_MACRO(!initialized, ERR_NOT_INITIALIZED, 0);
 
     while (((volatile Sound_Sample *) sample_list) != NULL)
-        Sound_FreeSample(sample_list);
+    {
+        Sound_Sample *sample = sample_list;
+        Sound_FreeSample(sample); /* Updates sample_list. */
```

Comment:
Wow - so `Sound_FreeSample` was cutting the legs out from under the `sample_list` content even before the while loop could get through all the items.

Reply:
Oh, no - the implementation is completely fine and does what it is supposed to do (removes 1 element from global doubly-linked list sample_list). Coverity got confused and flagged an issue here, but it is a false-positive. I could've marked it as false-positive in Coverity database, but I hope this tiny workaround will prevent Coverity from flagging it.

Reply:
(sorry for you commenting twice on this; I commented on the code before reading your helpful commit message) I like your work around though to avoid operating on the global, and instead use a temporary. 

Longer term, it would be nice to move all of the SDL sound bits to smart objects; I don't think there's much left after we removed a bunch of code and intermediate buffers in the last big commit to it.